libs/auth-api-lib/src/lib/clients/clients.service.ts

```diff
@@ -620,17 +618,32 @@ export class ClientsService {
       )
     }
 
-    return Promise.all(
-      delegationTypes.map((delegationType) =>
-        this.clientDelegationType.upsert(
+    const upsertPromises = delegationTypes.map((delegationType) => {
```

Comment:
Instead of complicating this map you could do a check if `delegationTypes` includes `AuthDelegationType.Custom` just overwrite the `delegationTypes` with new array including the `GeneralMandate` (shallow copying the array just in case, so its not modifing the source array in place if it would cause issues in the source context)

```
if (delegationTypes.includes(AuthDelegationType.Custom)) {
  delegationTypes = [
    ...delegationTypes,
   AuthDelegationType.GeneralMandate
  ]
}

// Rest should be the same
```